lib/ecto/changeset.ex

```diff
@@ -1523,10 +1523,10 @@ defmodule Ecto.Changeset do
 
   defp cast_merge(cs1, cs2) do
     new_params = (cs1.params || cs2.params) && Map.merge(cs1.params || %{}, cs2.params || %{})
+    new_types = (cs1.types || cs2.types) && Map.merge(cs1.types || %{}, cs2.types || %{})
```

Comment:
From the `Ecto.Changeset.t/1` typespec, `types` can be `nil`, so I implemented fallbacks with `||`. However, in the context of `Ecto.Changeset.merge/2`, I'm not sure if `types` can actually be `nil`, since an `(ArgumentError) different :data when merging changesets` error is raised if the `data` fields differ.

Can someone confirm if it's safe to simply `Map.merge(cs1.types, cs2.types)` for a cleaner solution?